lib/TabTracker.js

```diff
@@ -33,30 +34,32 @@ TabTracker.prototype = {
   },
 
   getLastActivation: function(tab) {
-    if (!ss.storage.tabData[tab.url] || !ss.storage.tabData[tab.url][tab.id]) {
+    if (!this._tabData[tab.url] || !this._tabData[tab.url][tab.id]) {
       return null;
     }
-    let activations = ss.storage.tabData[tab.url][tab.id].activations;
+    let activations = this._tabData[tab.url][tab.id].activations;
     return activations[activations.length - 1];
   },
 
   navigateAwayFromPage: function(tab) {
-    if (!ss.storage.tabData[tab.url][tab.id]) {
+    if (!this._tabData[tab.url][tab.id]) {
       // We're navigating away from an activity streams page that
       // didn't even load yet. Let's say it's been active for 0 seconds.
-      ss.storage.tabData[tab.url][tab.id] = {activations: [{startTime: Date.now(), totalTime: 0}]};
+      this._tabData[tab.url][tab.id] = {activations: [{startTime: Date.now(), totalTime: 0}]};
+      DataSender.send(DataSender.TAB_SESSION_COMPLETE, this._tabData);
```

Comment:
Can you please send the tabData as part of an observer notification?

This gives us flexibility as far as how the data ends up in the data sender.